### PR TITLE
feat: Core data operation in the main thread

### LINKED ISSUE #2873: DB on the main thread: Core Data
Performing DB queries on the main thread can lead to performance issues. To enable a performance issue detector on the backend to detect such queries, extend existing DB instrumentation to report whether the query was executed on the main thread.

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Adding extra information to a core data span when running in the main thread

## :bulb: Motivation and Context

close #2873 

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/include/SentryNSProcessInfoWrapper.h [hunk 1]
```diff
@@ -4,6 +4,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface SentryNSProcessInfoWrapper : NSObject
 
+@property (class, readonly) SentryNSProcessInfoWrapper *shared;
```
Comment (armcknight) [inline Sources/Sentry/include/SentryNSProcessInfoWrapper.h:7]: Is there any way we can do what you need without introducing more singletons? Can you let us know why it's better/necessary to convert this to a singleton?

Comment (brustolin) [reply Sources/Sentry/include/SentryNSProcessInfoWrapper.h:7]: Yes its possible, by creating another instance of SentryNSProcessInfoWrapper.
`SentryNSProcessInfoWrapper` consist of 3 functions that pass forward the call to another API, it does not have any variables nor any behaviour. Why should we allocated memory to hold reference to it every time we need?

**Review by philipphofmann — COMMENTED** [review]
Almost LGTM, thanks @brustolin 👍

**Review by philipphofmann — APPROVED** [review]
LGTM 🎉

**Pull request merged**